stan/math/prim/mat/fun/divide_columns.hpp

```diff
@@ -49,17 +24,19 @@ inline typename std::vector<Eigen::Matrix<T_x, R, C>> divide_columns(
  *
  */
 template <typename T_x, typename T_v, int R, int C>
-inline typename std::vector<Eigen::Matrix<T_x, R, C>> divide_columns(
-    const std::vector<Eigen::Matrix<T_x, R, C>> &x,
-    const std::vector<T_v> &vec) {
+inline typename std::vector<
+    Eigen::Matrix<typename return_type<T_x, T_v, double>::type, R, C>>
+divide_columns(const std::vector<Eigen::Matrix<T_x, R, C>> &x,
+               const std::vector<T_v> &vec) {
   size_t N = x.size();
   size_t D = x[0].size();
   check_size_match("divide_columns", "x dimension", D, "vector", vec.size());
 
-  std::vector<Eigen::Matrix<T_x, R, C>> out(N);
+  std::vector<Eigen::Matrix<typename return_type<T_x, T_v, double>::type, R, C>>
```

Comment:
Yeah I think you're right here. This means that divide_columns was not working if the scalar types in x were doubles and the scalar types in vec were vars?

Reply:
something like that... I'm not looking at the code in depth right now, but I was playing with `T_x` parameter, in some cases it was a matrix, so it's better if I specifiy `T_x` and a scalar type and then set the dimensions as `R`, `C`. This is what compiled when I dropped it into command stan